src/thread.cpp

```diff
@@ -27,7 +27,8 @@
 
 using namespace Search;
 
-ThreadPool Threads; // Global object
+ThreadPool Threads;
+thread_local Thread *ThisThread;
```

Comment:
thread_local is a nice solution, but the last time I tested it, I found very slow:

http://stackoverflow.com/questions/506093/why-is-thread-local-storage-so-slow

BTW that's one of the reasons why thread is in position.